src/test/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidatorTest.java

```diff
@@ -0,0 +1,287 @@
+                ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Tests the validation fails on previous period incorrect total")
+    void testIncorrectPreviousDebtorsTotal() throws Throwable {
+
+        CompanyProfile companyProfile = addMultipleYearFilingCompany();
+
+        addValidCurrentDebtors();
+
+        PreviousPeriod previousDebtors = new PreviousPeriod();
+        previousDebtors.setTradeDebtors(2L);
+        previousDebtors.setTotal(20L);
+
+        debtors.setPreviousPeriod(previousDebtors);
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        ReflectionTestUtils.setField(validator, "incorrectTotal", "incorrect_total");
+
+        validator.isMultipleYearFiler(transaction);
+        errors = validator.validateDebtors(debtors, transaction);
+
+        assertTrue(errors.containsError(
+            new Error("incorrect_total", PREVIOUS_TOTAL_PATH,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Tests the validation fails on previous period missing total")
+    void testMissingPreviousDebtorsTotal() throws Throwable {
+
+        CompanyProfile companyProfile = addMultipleYearFilingCompany();
+
+        addValidCurrentDebtors();
+
+        PreviousPeriod previousDebtors = new PreviousPeriod();
+        previousDebtors.setTradeDebtors(2L);
+
+        debtors.setPreviousPeriod(previousDebtors);
+
+        when(transaction.getCompanyNumber()).thenReturn("12345");
+        when(environmentReader.getMandatoryString(any())).thenReturn("http://test-url");
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenReturn(companyProfile);
+
+        ReflectionTestUtils.setField(validator, "invalidNote", "invalid_note");
+
+        validator.isMultipleYearFiler(transaction);
+        errors = validator.validateDebtors(debtors, transaction);
+
+        Mockito.verify(restTemplate, Mockito.atLeastOnce()).getForObject(anyString(), any(Class.class));
+
+        assertTrue(errors.containsError(
+            new Error("invalid_note", PREVIOUS_TOTAL_PATH,
+                LocationType.JSON_PATH.getValue(),
+                ErrorType.VALIDATION.getType())));
+
+    }
+
+    @Test
+    @DisplayName("Tests Rest exception is thrown when company-profile not found")
+    void testRestExceptionThrownWhenApiCallFails() {
+
+        addValidCurrentDebtors();
+
+        when(restTemplate.getForObject(anyString(), any(Class.class))).thenThrow(restClientException);
+
+        Executable executable = () -> {
+            validator.isMultipleYearFiler(transaction);
+        };
+        assertThrows(RestException.class, executable);
+    }
+
+
+    private void addValidCurrentDebtors() {
```

Comment:
This method could be moved below the tests so that the private methods are grouped below